src/pybind/mgr/rbd_support/trash_purge_schedule.py

```diff
@@ -0,0 +1,495 @@
+import errno
+import json
+import rados
+import rbd
+import re
+import traceback
+
+from datetime import datetime
+from threading import Condition, Lock, Thread
+
+from .common import get_rbd_pools
+from .schedule import Interval, StartTime, Schedule
+
+SCHEDULE_OID = "rbd_trash_trash_purge_schedule"
+
+SCHEDULE_INTERVAL = "interval"
+SCHEDULE_START_TIME = "start_time"
+
+
+class LevelSpec:
```

Comment:
Nit: might be nice to combine the mirror and trash `LevelSpec` into a single class w/ an optional parameter to indicate if it supports image names.